doc/api/process.md

```diff
@@ -1684,6 +1684,107 @@ the [`'warning'` event][process_warning] and the
 [`emitWarning()` method][process_emit_warning] for more information about this
 flag's behavior.
 
+## `process.parseArgs([argv[, options]])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `argv` {Array<string>|Object} (Optional) Array of argument strings; defaults
+  to [`process.argv.slice(2)`](process_argv). If an Object, the default is used,
+  and this parameter is considered to be the `options` parameter.
+* `options` {Object} (Optional) The `options` parameter, if present, is an
+  object supporting the following property:
+  * `expectsValue` {Array<string>|string} (Optional) One or more argument
+    strings which _expect a value_ when present
+* Returns: {Object} An object having properties corresponding to parsed Options
+  and Flags, and a property `_` containing Positionals
+
+The `process.parseArgs` function parses command-line arguments from an array of
+strings and returns an object representation.
```

Comment:
I think these docs should clarify that this is not a parser for the *Node.js* command line flags, and instead a generic one

Reply:
Thinking more about it, I think the reason that I expected this to be a Node.js command line flag parser was that it’s added to `process`. I don’t think that’s a good place for a generic function, and I think `util.parseArgs()` would be a better choice (and I would not expect that to be controversial, either)

Reply:
Perhaps `util.parseArgs` method and a `process.parsedArgs` getter? `process.argv.slice(2)` is an incantation worth avoiding.

Reply:
I am fine with moving it into `util` if that's the general consensus

Reply:
I also think `util` is more appropriate than `process`

Reply:
+1 on moving to `util`